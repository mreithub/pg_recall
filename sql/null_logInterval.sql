--
-- Creates a managed table with a NULL log interval and makes sure it's not affected 
-- by cleanup() calls
--

BEGIN;

CREATE TABLE config (
	key VARCHAR(100) PRIMARY KEY,
	value TEXT NOT NULL
);

SELECT recall.enable('config', NULL);


-- run some statements
INSERT INTO config VALUES ('foo', 'bar');
INSERT INTO config VALUES ('true', false);

-- 'wait' an hour
SELECT pretendToWait('1 hour');

-- run some more statements
INSERT INTO config VALUES ('answer', 42);
UPDATE config SET value=true WHERE key='true';


-- cleanup (should return 0)
SELECT recall.cleanup('config');

-- and check the log data (there should be 4 rows)
SELECT key, value, UPPER(_log_time) IS NULL AS is_current FROM recall.config_log;

ROLLBACK;
