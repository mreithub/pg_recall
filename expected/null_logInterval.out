--
-- Creates a managed table with a NULL log interval and makes sure it's not affected 
-- by cleanup() calls
--
BEGIN;
CREATE TABLE config (
	key VARCHAR(100) PRIMARY KEY,
	value TEXT NOT NULL
);
SELECT recall.enable('config', NULL);
 enable 
--------
 
(1 row)

-- run some statements
INSERT INTO config VALUES ('foo', 'bar');
INSERT INTO config VALUES ('true', false);
-- 'wait' an hour
SELECT pretendToWait('1 hour');
 pretendtowait 
---------------
 
(1 row)

-- run some more statements
INSERT INTO config VALUES ('answer', 42);
UPDATE config SET value=true WHERE key='true';
-- cleanup (should return 0)
SELECT recall.cleanup('config');
NOTICE:  recall: Cleaning up table config
 cleanup 
---------
       0
(1 row)

-- and check the log data (there should be 4 rows)
SELECT key, value, UPPER(_log_time) IS NULL AS is_current FROM recall.config_log;
  key   | value | is_current 
--------+-------+------------
 foo    | bar   | t
 answer | 42    | t
 true   | false | f
 true   | true  | t
(4 rows)

ROLLBACK;
